--- src/Autoload/ScoperAutoloadGenerator.php	2025-06-04 20:36:51
+++ src/Autoload/ScoperAutoloadGenerator.php	2025-06-04 20:37:31
@@ -108,7 +108,7 @@
                         \$loader = require_once __DIR__.'/autoload.php';
                         // Ensure InstalledVersions is available
                         \$installedVersionsPath = __DIR__.'/composer/InstalledVersions.php';
-                        if (file_exists(\$installedVersionsPath)) require_once \$installedVersionsPath;
+                        if (!class_exists(\Composer\InstalledVersions::class, false) && file_exists(\$installedVersionsPath)) require_once \$installedVersionsPath;

                         // Restore the backup and ensure the excluded files are properly marked as loaded
                         \$GLOBALS['__composer_autoload_files'] = \\array_merge(
@@ -140,7 +140,7 @@
                     \$loader = require_once __DIR__.'/autoload.php';
                     // Ensure InstalledVersions is available
                     \$installedVersionsPath = __DIR__.'/composer/InstalledVersions.php';
-                    if (file_exists(\$installedVersionsPath)) require_once \$installedVersionsPath;
+                    if (!class_exists(\Composer\InstalledVersions::class, false) && file_exists(\$installedVersionsPath)) require_once \$installedVersionsPath;

                     // Restore the backup and ensure the excluded files are properly marked as loaded
                     \$GLOBALS['__composer_autoload_files'] = \\array_merge(
