--- packages/PHPStanStaticTypeMapper/TypeMapper/BooleanTypeMapper.php	2022-09-23 11:46:53.000000000 +0200
+++ packages/PHPStanStaticTypeMapper/TypeMapper/BooleanTypeMapper.php	2022-09-27 11:04:44.000000000 +0200
@@ -44,7 +44,7 @@
         }
         if ($type instanceof ConstantBooleanType) {
             // cannot be parent of union
-            return new IdentifierTypeNode('true');
+            return new IdentifierTypeNode('false');
         }
         return new IdentifierTypeNode('bool');
     }
