--- Core/Core.php
+++ Core/Core.php
@@ -660,8 +660,6 @@
  * </p>
  * @return bool This function returns false if wrong <i>error_type</i> is
  * specified, true otherwise.
- * @throws \ValueError This function throws a ValueError if error_level is not one of E_USER_ERROR,
- * E_USER_WARNING, E_USER_NOTICE, E_USER_DEPRECATED.
  */
 #[LanguageLevelTypeAware(['8.4' => 'true'], default: 'bool')]
 function trigger_error(string $message, int $error_level = E_USER_NOTICE) {}
--- standard/standard_2.php
+++ standard/standard_2.php
@@ -337,13 +337,6 @@
  * </p>
  * @return string a string produced according to the formatting string
  * format.
- * @throws \ValueError As of PHP 8.0.0, a ValueError is thrown if the number of arguments is zero.
- * Prior to PHP 8.0.0, a E_WARNING was emitted instead. As of PHP 8.0.0, a ValueError is thrown if
- * [width] is less than zero or bigger than PHP_INT_MAX. Prior to PHP 8.0.0, a E_WARNING was emitted
- * instead. As of PHP 8.0.0, a ValueError is thrown if [precision] is less than zero or bigger than
- * PHP_INT_MAX. Prior to PHP 8.0.0, a E_WARNING was emitted instead.
- * @throws \ArgumentCountError As of PHP 8.0.0, a ArgumentCountError is thrown when less arguments
- * are given than required. Prior to PHP 8.0.0, false was returned and a E_WARNING emitted instead.
  */
 #[Pure]
 function sprintf(
