--- dom/dom_c.php	2024-01-02 12:04:54
+++ dom/dom_c.php	2024-01-21 10:41:56
@@ -1347,6 +1347,14 @@
  */
 class DOMNamedNodeMap implements IteratorAggregate, Countable
 {
+
+     /**
+      * The number of nodes in the map. The range of valid child node indices is 0 to length - 1 inclusive.
+      * @var int
+      * @readonly
+      */
+    #[PhpStormStubsElementAvailable(from: '8.1')]
+    public $length;
     /**
      * Retrieves a node specified by name
      * @link https://php.net/manual/en/domnamednodemap.getnameditem.php
