--- dom/dom_c.php	2024-01-02 12:04:54
+++ dom/dom_c.php	2024-01-21 10:41:56
@@ -1347,6 +1347,14 @@
  */
 class DOMNamedNodeMap implements IteratorAggregate, Countable
 {
+
+     /**
+      * The number of nodes in the map. The range of valid child node indices is 0 to length - 1 inclusive.
+      * @var int
+      * @readonly
+      */
+    #[PhpStormStubsElementAvailable(from: '8.1')]
+    public $length;
     /**
      * Retrieves a node specified by name
      * @link https://php.net/manual/en/domnamednodemap.getnameditem.php
@@ -1779,7 +1787,7 @@
      * A <classname>DOMNamedNodeMap</classname> containing the attributes of this node (if it is a <classname>DOMElement</classname>) or NULL otherwise.
      * @link https://php.net/manual/en/class.domnode.php#domnode.props.attributes
      */
-    #[LanguageLevelTypeAware(['8.1' => 'DOMNamedNodeMap|null'], default: '')]
+    #[LanguageLevelTypeAware(['8.1' => 'DOMNamedNodeMap'], default: '')]
     public $attributes;

     /**
