diff --git a/Grammar.pp b/Grammar.pp
index 5157084..8384417 100644
--- a/Grammar.pp
+++ b/Grammar.pp
@@ -73,7 +73,7 @@
 %token  co:comment               .*?(?=(?<!\\)\))

 // Capturing group.
-%token  named_capturing_         \(\?<              -> nc
+%token  named_capturing_         \(\?P?<            -> nc
 %token  nc:_named_capturing      >                  -> default
 %token  nc:capturing_name        .+?(?=(?<!\\)>)
 %token  non_capturing_           \(\?:
@@ -109,7 +109,7 @@
 // Please, see PCRESYNTAX(3), General Category properties, PCRE special category
 // properties and script names for \p{} and \P{}.
 %token character_type            \\([CdDhHNRsSvVwWX]|[pP]{[^}]+})
-%token anchor                    \\(bBAZzG)|\^|\$
+%token anchor                    \\([bBAZzG])|\^|\$
 %token match_point_reset         \\K
 %token literal                   \\.|.

@@ -168,7 +168,7 @@ quantifier:
         ::negative_class_:: #negativeclass
       | ::class_::
     )
-    ( range() | literal() )+
+    ( <class_> | range() | literal() )+
     ::_class::

 #range:
@@ -178,7 +178,7 @@ simple:
     capturing()
   | literal()

-capturing:
+#capturing:
     ::comment_:: <comment>? ::_comment:: #comment
   | (
         ::named_capturing_:: <capturing_name> ::_named_capturing:: #namedcapturing
@@ -191,6 +191,7 @@ capturing:

 literal:
     <character>
+  | <range>
   | <dynamic_character>
   | <character_type>
   | <anchor>
