--- packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2023-01-06 14:56:29.000000000 +0100
+++ packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2023-01-06 14:57:51.000000000 +0100
@@ -14,6 +14,7 @@
 use PHPStan\Type\FloatType;
 use PHPStan\Type\IntegerType;
 use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
 use PHPStan\Type\ObjectWithoutClassType;
 use PHPStan\Type\StaticType;
 use PHPStan\Type\StringType;
@@ -102,7 +103,7 @@
             return new StaticType($classReflection);
         }
         if ($reference === ObjectReference::SELF) {
-            return new SelfStaticType($classReflection);
+            return new ObjectType($classReflection->getName());
         }
         if ($reference === ObjectReference::PARENT) {
             $parentClassReflection = $classReflection->getParentClass();
@@ -111,7 +112,7 @@
             }
             return new ParentObjectWithoutClassType();
         }
-        return new ThisType($classReflection);
+        return new ObjectType($classReflection->getName());
     }
     /**
      * @return \PHPStan\Type\ArrayType|\PHPStan\Type\IntegerType|\PHPStan\Type\FloatType|\PHPStan\Type\StringType|\PHPStan\Type\Constant\ConstantBooleanType|\PHPStan\Type\BooleanType|\PHPStan\Type\MixedType
