--- packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2023-07-23 22:44:23
+++ packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2023-07-25 09:45:33
@@ -13,6 +13,7 @@
 use PHPStan\Type\FloatType;
 use PHPStan\Type\IntegerType;
 use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
 use PHPStan\Type\ObjectWithoutClassType;
 use PHPStan\Type\StaticType;
 use PHPStan\Type\StringType;
@@ -92,7 +93,7 @@
             return new StaticType($classReflection);
         }
         if ($reference === ObjectReference::SELF) {
-            return new SelfStaticType($classReflection);
+            return new ObjectType($classReflection->getName());
         }
         if ($reference === ObjectReference::PARENT) {
             $parentClassReflection = $classReflection->getParentClass();
@@ -101,7 +102,7 @@
             }
             return new ParentObjectWithoutClassType();
         }
-        return new ThisType($classReflection);
+        return new ObjectType($classReflection->getName());
     }
     /**
      * @return \PHPStan\Type\ArrayType|\PHPStan\Type\IntegerType|\PHPStan\Type\FloatType|\PHPStan\Type\StringType|\PHPStan\Type\Constant\ConstantBooleanType|\PHPStan\Type\BooleanType|\PHPStan\Type\MixedType
