@package rector/rector

--- packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2021-11-23 18:38:29.000000000 +0100
+++ packages/StaticTypeMapper/PhpParser/NameNodeMapper.php	2021-12-16 23:09:30.000000000 +0100
@@ -106,7 +106,7 @@
             }
             return new \Rector\StaticTypeMapper\ValueObject\Type\ParentObjectWithoutClassType();
         }
-        return new \PHPStan\Type\ThisType($classReflection);
+        return new \PHPStan\Type\ObjectType($classReflection->getName());
     }
     /**
      * @return \PHPStan\Type\ArrayType|\PHPStan\Type\BooleanType|\PHPStan\Type\Constant\ConstantBooleanType|\PHPStan\Type\FloatType|\PHPStan\Type\IntegerType|\PHPStan\Type\MixedType|\PHPStan\Type\StringType
