@package rector/rector

--- config/phpstan/parser.neon	2022-01-05 19:37:59.000000000 +0100
+++ config/phpstan/parser.neon	2022-01-06 21:31:23.000000000 +0100
@@ -5,11 +5,18 @@
         factory: @pathRoutingParser
         arguments!: []

+    cachedRectorParser:
+        class: PHPStan\Parser\CachedParser
+        arguments:
+            originalParser: @rectorParser
+            cachedNodesByStringCountMax: %cache.nodesByStringCountMax%
+        autowired: false
+
     pathRoutingParser:
         class: PHPStan\Parser\PathRoutingParser
         arguments:
-            currentPhpVersionRichParser: @rectorParser
-            currentPhpVersionSimpleParser: @rectorParser
+            currentPhpVersionRichParser: @cachedRectorParser
+            currentPhpVersionSimpleParser: @cachedRectorParser
             php8Parser: @php8Parser
         autowired: false

