--- PDO/PDO.php	2021-12-26 15:44:39.000000000 +0100
+++ PDO/PDO.php	2022-01-03 22:54:21.000000000 +0100
@@ -1476,7 +1476,7 @@ namespace {
          * @return array|false if one or more notifications is pending, returns a single row,
          * with fields message and pid, otherwise <b>FALSE</b>.
          */
-        public function pgsqlGetNotify($fetchMode = PDO::FETCH_DEFAULT, $timeoutMilliseconds = 0) {}
+        public function pgsqlGetNotify($fetchMode = 1, $timeoutMilliseconds = 0) {}

         /**
          * (PHP 5 &gt;= 5.6.0, PHP 7, PHP 8)<br/>
