--- Read.php	2017-07-11 09:42:15
+++ Read.php	2024-08-26 23:09:54
@@ -77,7 +77,7 @@
      * @throws  \Hoa\File\Exception\FileDoesNotExist
      * @throws  \Hoa\File\Exception
      */
-    protected function &_open($streamName, Stream\Context $context = null)
+    protected function &_open($streamName, ?Stream\Context $context = null)
     {
         static $createModes = [
             parent::MODE_READ
