--- Reflection/ReflectionClass.php	2026-04-24 11:57:10
+++ Reflection/ReflectionClass.php	2026-04-25 21:47:47
@@ -305,7 +305,7 @@
      *
      * @link https://php.net/manual/en/reflectionclass.getconstants.php
      * @param int|null $filter [optional] allows the filtering of constants defined in a class by their visibility. Since 8.0.
-     * @return array<string, scalar|array<scalar>> An array of constants, where the keys hold the name and
+     * @return array<string, mixed> An array of constants, where the keys hold the name and
      * the values the value of the constants.
      */
     #[Pure]
@@ -317,7 +317,7 @@
      *
      * @link https://php.net/manual/en/reflectionclass.getconstant.php
      * @param string $name Name of the constant.
-     * @return scalar|array<scalar> Value of the constant with the name name.
+     * @return mixed Value of the constant with the name name.
      * Returns {@see false} if the constant was not found in the class.
      */
     #[Pure]
