--- session/SessionHandler.php	2021-11-04 14:27:30.000000000 +0100
+++ session/SessionHandler.php	2021-11-05 11:26:14.000000000 +0100
@@ -147,7 +147,7 @@
      * </p>
      */
     #[TentativeType]
-    public function validateId(string $id): bool;
+    public function validateId($id): bool;

     /**
      * Update timestamp of a session
@@ -163,7 +163,7 @@
      * @return bool
      */
     #[TentativeType]
-    public function updateTimestamp(string $id, string $data): bool;
+    public function updateTimestamp($id, $data): bool;
 }

 /**
