--- Stream.php	2024-08-26 23:05:49
+++ Stream.php	2024-08-26 23:01:08
@@ -192,7 +192,7 @@
      * @return  array
      * @throws  \Hoa\Stream\Exception
      */
-    final private static function &_getStream(
+    private static function &_getStream(
         $streamName,
         Stream $handler,
         $context = null
@@ -250,7 +250,7 @@
      * @return  resource
      * @throws  \Hoa\Exception\Exception
      */
-    abstract protected function &_open($streamName, Context $context = null);
+    abstract protected function &_open($streamName, ?Context $context = null);

     /**
      * Close the current stream.
@@ -687,11 +687,6 @@
 Consistency::flexEntity('Hoa\Stream\Stream');

 /**
- * Shutdown method.
- */
-Consistency::registerShutdownFunction(xcallable('Hoa\Stream\Stream::_Hoa_Stream'));
-
-/**
  * Add the `hoa://Library/Stream` node. Should be use to reach/get an entry
  * in the stream register.
  */
