@package rector/rector

--- packages/NodeTypeResolver/PHPStan/Type/TypeFactory.php	2021-11-23 18:38:29.000000000 +0100
+++ packages/NodeTypeResolver/PHPStan/Type/TypeFactory.php	2021-12-18 22:49:01.000000000 +0100
@@ -122,9 +122,6 @@
         if ($type instanceof \PHPStan\Type\Constant\ConstantIntegerType) {
             return new \PHPStan\Type\IntegerType();
         }
-        if ($type instanceof \PHPStan\Type\Constant\ConstantBooleanType) {
-            return new \PHPStan\Type\BooleanType();
-        }
         return $type;
     }
     /**
