--- src/Ast/PhpDoc/TypeAliasTagValueNode.php
+++ src/Ast/PhpDoc/TypeAliasTagValueNode.php
@@ -4,6 +4,7 @@
 
 use PHPStan\PhpDocParser\Ast\NodeAttributes;
 use PHPStan\PhpDocParser\Ast\Type\TypeNode;
+use function implode;
 use function trim;
 
 class TypeAliasTagValueNode implements PhpDocTagValueNode
@@ -15,15 +16,25 @@
 
 	public TypeNode $type;
 
-	public function __construct(string $alias, TypeNode $type)
+	/** @var TemplateTagValueNode[] */
+	public array $templateTypes;
+
+	/**
+	 * @param TemplateTagValueNode[] $templateTypes
+	 */
+	public function __construct(string $alias, TypeNode $type, array $templateTypes = [])
 	{
 		$this->alias = $alias;
 		$this->type = $type;
+		$this->templateTypes = $templateTypes;
 	}
 
 	public function __toString(): string
 	{
-		return trim("{$this->alias} {$this->type}");
+		$templateTypes = $this->templateTypes !== []
+			? '<' . implode(', ', $this->templateTypes) . '>'
+			: '';
+		return trim("{$this->alias}{$templateTypes} {$this->type}");
 	}
 
 	/**
@@ -31,7 +42,7 @@
 	 */
 	public static function __set_state(array $properties): self
 	{
-		$instance = new self($properties['alias'], $properties['type']);
+		$instance = new self($properties['alias'], $properties['type'], $properties['templateTypes'] ?? []);
 		if (isset($properties['attributes'])) {
 			foreach ($properties['attributes'] as $key => $value) {
 				$instance->setAttribute($key, $value);
