--- Core.php	2020-11-01 13:45:07.000000000 +0100
+++ Core2.php	2020-11-01 14:32:19.000000000 +0100
@@ -336,7 +336,7 @@
  * </p>
  * @return bool true on success or false on failure.
  */
-function define ($constant_name, $value, #[Deprecated] $case_insensitive = false) {}
+function define ($constant_name, $value, $case_insensitive = false) {}

 /**
  * Checks whether a given named constant exists
