--- PDO/PDO.stub	2020-06-14 14:26:12.000000000 +0200
+++ PDO/PDO2.stub	2020-06-14 14:26:12.000000000 +0200
@@ -843,6 +843,15 @@
 	 */
 	const SQLITE_ATTR_EXTENDED_RESULT_CODES = 2;

+	const FB_ATTR_DATE_FORMAT = 1;
+	const FB_ATTR_TIME_FORMAT = 2;
+	const FB_ATTR_TIMESTAMP_FORMAT = 3;
+
+	const OCI_ATTR_ACTION = 1;
+	const OCI_ATTR_CLIENT_INFO = 2;
+	const OCI_ATTR_CLIENT_IDENTIFIER = 3;
+	const OCI_ATTR_MODULE = 4;
+
 	/**
 	 * (PHP 5 &gt;= 5.1.0, PHP 7, PECL pdo &gt;= 0.1.0)<br/>
 	 * Creates a PDO instance representing a connection to a database
