--- PDO.stub	2019-12-05 17:56:26.000000000 +0100
+++ PDO2.stub	2020-05-26 14:16:32.000000000 +0200
@@ -268,7 +268,6 @@
 	 * As <b>PDO::FETCH_INTO</b> but object is provided as a serialized string.
 	 * Available since PHP 5.1.0. Since PHP 5.3.0 the class constructor is never called if this
 	 * flag is set.
-	 * @since 5.1 Available
 	 * @link https://php.net/manual/en/pdo.constants.php#pdo.constants.fetch-serialize
 	 */
 	const FETCH_SERIALIZE = 524288;
@@ -727,7 +726,7 @@
 	 * <p>
 	 * Note, this constant can only be used in the driver_options array when constructing a new database handle.
 	 * </p>
-	 * @since 5.5.21 and 5.6.5
+	 * @since 5.5.21
 	 * @link https://php.net/manual/en/ref.pdo-mysql.php#pdo.constants.mysql-attr-multi-statements
 	 */
 	const MYSQL_ATTR_MULTI_STATEMENTS = 1015;
@@ -747,6 +746,7 @@
 	 */
 	const PGSQL_ASSOC = 1;
 	const PGSQL_ATTR_DISABLE_NATIVE_PREPARED_STATEMENT = 1000;
+	const PGSQL_ATTR_DISABLE_PREPARES = 1000;
 	const PGSQL_BAD_RESPONSE = 5;
 	const PGSQL_BOTH = 3;
 	const PGSQL_TRANSACTION_IDLE = 0;
@@ -839,6 +839,15 @@
 	 */
 	const SQLITE_ATTR_EXTENDED_RESULT_CODES = 2;
 
+	const FB_ATTR_DATE_FORMAT = 1;
+	const FB_ATTR_TIME_FORMAT = 2;
+	const FB_ATTR_TIMESTAMP_FORMAT = 3;
+
+	const OCI_ATTR_ACTION = 1;
+	const OCI_ATTR_CLIENT_INFO = 2;
+	const OCI_ATTR_CLIENT_IDENTIFIER = 3;
+	const OCI_ATTR_MODULE = 4;
+
 	/**
 	 * (PHP 5 &gt;= 5.1.0, PHP 7, PECL pdo &gt;= 0.1.0)<br/>
 	 * Creates a PDO instance representing a connection to a database
